src/main/java/soolutions/hometaskapp/core/Amount.java

```diff
@@ -0,0 +1,40 @@
+package soolutions.hometaskapp.core;
+
+// TODO(vivek): Write docs for each public class
+public final class Amount {
+  private final double value;
+  private final String currency;
+
+// TODO(vivek): Check if there is an exisiting library for currency handling
+  public Amount(double value, String currency) {
+    if (Double.MAX_VALUE < value || value < 0) {
```

Comment:
Why can't an amount be negative?

How could a supplied double value be greater than the max?

Should we be using floating precision numbers for a money structure?

Reply:
An `amount` in itself is not negative. For double you raise a valid point.

Reply:
I also don't see any test for this type of exception (`InvalidAmountException`)  so I think don't think we should merge this code without test coverage.